src/Features/Core/Portable/GenerateMember/GenerateConstructor/AbstractGenerateConstructorService.Editor.cs

```diff
@@ -286,14 +298,14 @@ private async Task<(Document, bool addedFields)> GenerateFieldDelegatingConstruc
                     // manner, then case insensitively.
                     if (!TryFindMatchingField(
                             arguments, attributeArguments, parameterNames, parameterTypes, i, parameterToExistingFieldMap,
-                            parameterToNewFieldMap, caseSensitive: true, newParameterNames: out parameterNames))
+                            parameterToNewFieldMap, caseSensitive: true, newParameterNames: out parameterNames, fieldNamingRule))
                     {
                         if (!TryFindMatchingField(
                                 arguments, attributeArguments, parameterNames, parameterTypes, i, parameterToExistingFieldMap,
-                                parameterToNewFieldMap, caseSensitive: false, newParameterNames: out parameterNames))
+                                parameterToNewFieldMap, caseSensitive: false, newParameterNames: out parameterNames, fieldNamingRule))
                         {
-                            parameterToNewFieldMap[parameterNames[i].BestNameForParameter] =
-                                parameterNames[i].NameBasedOnArgument;
+                            // If no matching field was found, use the fieldNamingRule to create suitable name
+                            parameterToNewFieldMap[parameterNames[i].BestNameForParameter] = fieldNamingRule.NamingStyle.MakeCompliant(parameterNames[i].NameBasedOnArgument).First();
```

Comment:
can you keep the wrapping here?  this can't be read well on github.

Reply:
how do we know 'First' will succeed?

Reply:
I believe `MakeCompliant` will always return at least one name: http://source.roslyn.io/#Microsoft.CodeAnalysis.Workspaces/NamingStyles/NamingStyle.cs,307